pkgs/development/python-modules/salmon-mail/default.nix

```diff
@@ -1,22 +1,22 @@
-{ stdenv, buildPythonPackage, fetchPypi, nose, dnspython
-,  chardet, lmtpd, python-daemon, six, jinja2, mock }:
+{ stdenv, buildPythonPackage, fetchPypi, dnspython, chardet, lmtpd
+, python-daemon, six, jinja2, mock, click }:
 
 buildPythonPackage rec {
   pname = "salmon-mail";
-  version = "3.1.1";
+  version = "3.2.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0ddd9nwdmiibk3jaampznm8nai5b7zalp0f8c65l71674300bqnw";
+    sha256 = "0q2m6xri1b7qv46rqpv2qfdgk2jvswj8lpaacnxwjna3m685fhfx";
   };
 
-  checkInputs = [ nose jinja2 mock ];
-  propagatedBuildInputs = [ chardet dnspython lmtpd python-daemon six ];
+  checkInputs = [ jinja2 mock ];
+  propagatedBuildInputs = [ chardet dnspython lmtpd python-daemon six click ];
 
   # The tests use salmon executable installed by salmon itself so we need to add
   # that to PATH
   checkPhase = ''
-    PATH=$out/bin:$PATH nosetests .
+    PATH=$out/bin:$PATH python setup.py test
```

Comment:
getting some failures:
```
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
builder for '/nix/store/446ahn0w3b5b7dqh1245hc55pcfsxbvj-python3.7-salmon-mail-3.2.0.drv' failed with exit code 1; last 10 log lines:
    File "/build/salmon-mail-3.2.0/salmon/routing.py", line 503, in routing_wrapper
      next_state = func(message, *args, **kw)
  TypeError: END() got an unexpected keyword argument 'anything'

  ----------------------------------------------------------------------
  Ran 147 tests in 2.561s

  FAILED (errors=1)
  Test failed: <unittest.runner.TextTestResult run=147 errors=1 failures=0>
  error: Test failed: <unittest.runner.TextTestResult run=147 errors=1 failures=0>
builder for '/nix/store/bl0xb0l6dmvkw46cq75dw5axzan75cxn-python3.8-salmon-mail-3.2.0.drv' failed with exit code 1; last 10 log lines:
    File "/build/salmon-mail-3.2.0/salmon/routing.py", line 503, in routing_wrapper
      next_state = func(message, *args, **kw)
  TypeError: END() got an unexpected keyword argument 'anything'

  ----------------------------------------------------------------------
  Ran 147 tests in 2.294s

  FAILED (errors=1)
  Test failed: <unittest.runner.TextTestResult run=147 errors=1 failures=0>
  error: Test failed: <unittest.runner.TextTestResult run=147 errors=1 failures=0>
cannot build derivation '/nix/store/j798yif7088svypcji7mzqqywq71dyw4-env.drv': 2 dependencies couldn't be built
```

Reply:
Well, that's weird.. This builds fine on my machine:
```
$ nix-build '<nixpkgs>' -A python37Packages.salmon-mail
/nix/store/p701xnxwrwvhg8pc76m02z29923lgxv1-python3.7-salmon-mail-3.2.0
```
What could cause the difference? Difficult to fix if I'm not able to reproduce..

Reply:
I now rebased on `nixos-unstable`. Still building fine on my machine...

Reply:
I'll come in as a neutral third party and say this builds just fine on my machine as well:

```
[4 built, 7 copied (2.1 MiB), 0.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/82166
2 package built:
python37Packages.salmon-mail python38Packages.salmon-mail
```

That said, the OfBorg build is failing on Darwin: 

```
 start explode localhost
EXPLODE!
Relay: <Relay to (localhost:0)>
error: Test failed: <unittest.runner.TextTestResult run=147 errors=0 failures=1>
builder for '/nix/store/dwdzz3r1j5abi09g48x6mhf5r406gy4r-python3.7-salmon-mail-3.2.0.drv' failed with exit code 1
```

Reply:
Hmm.. Should I try if `__darwinAllowLocalNetworking = true;` fixes build on Darwin?

Reply:
I now added that flag. Let's see if it helps. How can I use OfBorg to try building again for Darwin to see if it helped?